client/src/pages/metadata-query/freq-query.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useState } from 'react';
+
+import BandSelection from './band-selection';
+
+export const FreqQuery = ({ description, validator, queryName, handleQueryValid, handleQueryInvalid }) => {
+  const bands = {
+    VLF: ['VLF', 3000, 30000],
+    LF: ['LF', 30000, 3000000],
+    MF: ['MF', 300000, 3000000],
+    HF: ['HF', 3000000, 30000000],
+    VHF: ['VHF', 30000000, 300000000],
+    UHF: ['UHF', 300000000, 3000000000],
+  };
+  const [show, setShow] = useState(true);
+  const [freqRange, setFreqRange] = useState({
+    from: bands.VHF[1],
+    to: bands.VHF[2],
+  });
+  const [band, setBand] = useState(bands.VHF);
+
+  const handleFreqChange = (e) => {
+    const name = e.target.name;
+    const value = e.target.value;
+    const newfreqRange = { ...freqRange };
+    newfreqRange[name] = value;
+    setFreqRange(newfreqRange);
+    let valid =
+      name === 'from' ? validator({ from: value, to: freqRange.to }) : validator({ from: freqRange.from, to: value });
+    if (valid) {
+      return handleQueryValid(queryName, valid);
+    }
+    return handleQueryInvalid(queryName);
+  };
+
+  const renderDividerButtonClass = () => {
+    if (validator({ from: freqRange.from, to: freqRange.to })) {
+      return 'btn btn-success w-80';
+    }
+    return 'btn w-80';
+  };
+
+  const handleSelection = (to) => {
+    setBand(to);
+    setFreqRange({
+      from: to[1],
+      to: to[2],
+    });
+    const valid = validator({ from: to[1], to: to[2] });
+    return handleQueryValid(queryName, valid);
+  };
+
+  return (
+    <div className="mb-10">
+      <div className="divider mb-8">
+        <div className="tooltip" data-tip={description}>
+          <button
+            disabled={!validator({ from: freqRange.from, to: freqRange.to })}
+            onClick={() => setShow(!show)}
+            className={renderDividerButtonClass()}
+          >
+            {queryName}
+          </button>
+        </div>
+      </div>
+      {show && (
+        <div className="card bg-neutral text-neutral-content">
+          <div className="card-body">
+            <BandSelection selected={band[0]} handleSelection={handleSelection} bands={bands} />
+            <h3 className="text-lg">
+              From:
+              <span className="ml-2 badge badge-md badge-success">{`${freqRange.from} Hz`}</span>
```

Comment:
Why not use the dual range slider?

Reply:
This could definetly be improved to a dual range slider but in order to not make this pr even bigger than it already is I would add this as an improvement we could do later on.